### PR TITLE
feat(kuma-cp): define clusterName for Mesh*Service resources

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review

For a Mesh*Service I've chosen following naming convention 

`mesh_name_namespace_zone_msvc/mextsvc/mzsvc_port`

if there is no namespace or zone, value has an empty place `default_backend___msvc_8080`

- [ ] [Link to relevant issue][1] as well as docs and UI issues -- fix: https://github.com/kumahq/kuma/issues/11292
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/core/resources/model/resource.go [hunk 1]
```diff
@@ -718,6 +719,19 @@ func TargetRefToResourceIdentifier(meta ResourceMeta, tr common_api.TargetRef) R
...
 }
 
+func ResourceToBackendRef(r Resource, resType ResourceType, port uint32) common_api.BackendRef {
+	id := NewResourceIdentifier(r)
+	return common_api.BackendRef{
+		TargetRef: common_api.TargetRef{
+			Kind:      common_api.TargetRefKind(resType),
+			Name:      id.Name,
+			Namespace: id.Namespace,
+			Mesh:      id.Mesh,
```
Comment (lobkovilya) [inline pkg/core/resources/model/resource.go:729]: `Mesh` is reserved for cross-mesh, I doubt we actually take into account this value

File: pkg/xds/generator/zoneproxy/destinations.go [hunk 2]
```diff
@@ -63,12 +70,41 @@ func buildMeshServiceDestinations(
 				Mesh:            ms.GetMeta().GetMesh(),
 				DestinationName: ms.DestinationName(port.Port),
 				SNI:             sni,
+				Resource: model.ResolveBackendRef(
+					ms.Meta,
+					model.ResourceToBackendRef(ms, meshservice_api.MeshServiceType, port.Port),
+				),
```
Comment (lobkovilya) [inline pkg/xds/generator/zoneproxy/destinations.go:76]: If there is already a resource `ms`, why do we need to create BackendRef and then resolve it back to the resource? Can we just do:
```suggestion
				Resource: core_model.ResolvedBackendRef{
					Resource: pointer.To(core_model.NewTypedResourceIdentifier(ms, core_model.WithSectionName(port.GetName()))),
				},
```

File: pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go [hunk 3]
```diff
@@ -31,16 +32,6 @@ const (
 	HostnameGeneratorType MatchType = "HostnameGenerator"
 )
 
-// +kubebuilder:validation:Enum=tcp;grpc;http;http2
```
[the remark this answers is not in the document]
Comment (lukidzi) [reply pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go:34]: We are using `core_mesh.Protocol` in `MeshService` that's why I thought would be good to keep it the same across models

[the remark this answers is not in the document]
Comment (lobkovilya) [reply pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go:34]: I see, well it's not ideal, but moving `core_mesh.Protocol` to `common_api` definitely should be in a different PR.

BTW you can add `// +kubebuilder:validation:Enum=tcp;grpc;http;http2` right on the field (not on the type)

**Comment by lukidzi** [comment]
https://github.com/kumahq/kuma/pull/11398

**Pull request closed without merging**